engine/lib/comments.php

```diff
@@ -231,12 +231,22 @@ function _elgg_comments_container_permissions_override($hook, $type, $return, $p
 function _elgg_comments_permissions_override($hook, $type, $return, $params) {
 	$entity = $params['entity'];
 	$user = $params['user'];
-	
-	if (elgg_instanceof($entity, 'object', 'comment') && $user) {
-		return $entity->getOwnerGUID() == $user->getGUID();
+
+	if (!elgg_instanceof($entity, 'object', 'comment')) {
+		return $return;
 	}
-	
-	return $return;
+
+	if ($entity->owner_guid == $user->guid) {
```

Comment:
I've just tested what happens without it and without these lines the owner is NOT able to edit/delete the comment (as the handler does return false). As opposed to before the handler does no longer return canedit().